platoon/channel/controller.py

```diff
@@ -0,0 +1,554 @@
+
+    @staticmethod
+    def get_workers_devices(devices):
+        # Get Controller's devices
+        import socket
+        hostname = socket.gethostname()
+
+        if devices:
+            # 1. Use device names from arguments if they are specified
+            devices_found = devices
+        else:
+            # 2. Try device names from configuration
+            from platoon import configparser
+            try:
+                devices_found = configparser.fetch_devices_for_host(hostname)
+            except KeyError:
+                # 3. Else try to use all compatible GPUs in host
+                try:
+                    print("WARNING! Using all compatible GPUs in host.", file=sys.stderr)
+                    from pygpu import gpuarray as ga
+                    devcount = ga.count_devices("cuda", 0)
+                    print("WARNING! Found {} GPUs!".format(devcount), file=sys.stderr)
+                    devices_found = ["cuda" + str(i) for i in range(devcount)]
+                except ImportError:
+                    print("ERROR! Could not fetch devices for Controller.", file=sys.stderr)
+                    sys.exit(2)
+                except Exception as exc:
+                    print("ERROR! pygpu: {}".format(exc), file=sys.stderr)
+                    print("ERROR! Could not fetch devices for Controller.", file=sys.stderr)
+                    sys.exit(2)
+
+        if not devices_found:
+            print("ERROR! Cound not find any compatible GPUs in host.", file=sys.stderr)
+            sys.exit(4)
+
+        print("## On " + hostname + " using: " + " ".join(devices_found))
+        return devices_found
+
+    @staticmethod
+    def default_parser():
+        parser = argparse.ArgumentParser(
+            description="Base Platoon Controller process. Reigns over a computer node.")
+        parser.add_argument('experiment_name', help='The name of your experiment. The launcher will expect to find the files <experiment_name>_worker.py and optionally <experiment_name>_controller.py.')
+        parser.add_argument('log_directory', help='Directory where logging info and error files will exist.')
+        single_or_multi = parser.add_mutually_exclusive_group(required=True)
+        single_or_multi.add_argument('--single', action='store_true',
+                                     help='Indicates that this Controller participates in a single-node platoon.')
+        single_or_multi.add_argument('--multi', action='store_true',
+                                     help='Indicates that this Controller participates in a multi-node platoon. Requires mpi4py')
+        parser.add_argument('-D', '--devices', default=list(), nargs='+', type=str, metavar='devname',
+                            required=False, help='List of device names (e.g. gpu0 or cuda1). Each device will be assigned to a separate worker. If this option is specified, experiment will be run in a single node.')
```

Comment:
But gpu0 is still usable with the `param_sync` interface, isn't it? I will write a notice in README about compatibity issues.